internal/app/transport/request_test.go

```diff
@@ -0,0 +1,144 @@
+package transport
+
+import (
+	"testing"
+
+	discoveryv2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	corev2 "github.com/envoyproxy/go-control-plane/envoy/api/v2/core"
+	corev3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	discoveryv3 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/genproto/googleapis/rpc/status"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+const (
+	nodeID          = "1"
+	resourceName    = "route"
+	resourceVersion = "version"
+	cluster         = "cluster"
+	region          = "region"
+	zone            = "zone"
+	subzone         = "subzone"
+)
+
+var requestV2 = discoveryv2.DiscoveryRequest{
```

Comment:
May be better to structure the v2 tests in `requestv2_test.go` and the v3 ones in `requestv3_test.go` since you made the refactor here to separate the implementation from the interface.

Reply:
I think i prefer this structure. Imagine this is the test for the interface and it ties the behavior of all impl together in one place. Let me know if you still recommend to make the change.